sycl/test-e2e/format.py

```diff
@@ -54,8 +54,56 @@ def parse_min_intel_driver_req(line_number, line, output):
 
     return output
 
+# List of available features that affect building
+class FeatureInfo:
+    def __init__(self,
+                 device_agnostic = False,
+                 triples_always_has = set(),
+                 triples_never_has = set()):
+        self.device_agnostic = device_agnostic
+        self.triples_always_has = triples_always_has
+        self.triples_never_has = triples_never_has
+
+def getPossibleFeatures():
+    return {
+            "linux"                 :FeatureInfo(device_agnostic=True),
+            "system-linux"          :FeatureInfo(device_agnostic=True),
+            "windows"               :FeatureInfo(device_agnostic=True),
+            "system-windows"        :FeatureInfo(device_agnostic=True),
+            "build-and-run-mode"    :FeatureInfo(device_agnostic=True),
+            "opencl-aot"            :FeatureInfo(device_agnostic=True),
+            "ocloc"                 :FeatureInfo(device_agnostic=True),
+            "opencl_icd"            :FeatureInfo(device_agnostic=True),
+            "cl_options"            :FeatureInfo(device_agnostic=True),
```

Comment:
imo we have a relatively few number of tests that fail because of the triple or device stuff. imo supporting envrionment stuff like the OS is important, i think its fine to force those few number of tests require the build and run mode

having a hardcoded list of optioons and having different support for logical expressions than normal LIT kind of worries me